api/src/couchdb/attachmentExport.ts

```diff
@@ -0,0 +1,503 @@
+     * - filename: Original filename (unused in current implementation)
+     * - file_type: MIME type for determining file extension
+     */
+    const attachmentList = record.data[fieldId] as
+      | {
+          attachment_id: string;
+          filename: string;
+          file_type: string;
+        }[]
+      | null;
+
+    // Null indicates an incomplete or pending attachment field
+    if (attachmentList === null) {
+      console.log(`[ZIP] Skipping null attachment list in field ${fieldId}`);
+      continue;
+    }
+
+    // Process each attachment in the field
+    for (const {attachment_id, file_type} of attachmentList) {
+      logMemory(
+        'PROCESSING',
+        `Record ${recordCount} (${record.record_id}): ${attachment_id}`
+      );
+
+      // Generate a unique, collision-free filename for the archive
+      const fullFileName = generateFilenameForAttachment({
+        filenames,
+        fileMimeType: file_type,
+        hrid: record.hrid ?? record.record_id,
+        fieldId,
+        viewID: includeViewIDPrefix ? viewID : undefined,
+      });
+
+      // Track this filename to prevent duplicates
+      filenames.push(fullFileName);
+
+      /**
+       * Stream the attachment directly from database into the archive.
+       * This avoids loading the entire file into memory.
+       *
+       * The database returns a readable stream that we pipe into the archiver.
+       */
+      const fileReadStream = nanoDb.attachment.getAsStream(
+        attachment_id,
+        attachment_id
+      );
+
+      // Add the stream to the archive with the generated filename
+      archive.append(fileReadStream, {name: fullFileName});
+
+      /**
+       * Create a promise that resolves when the stream completes.
+       * These promises are returned to the caller for concurrency management.
+       */
+      const streamPromise = new Promise<void>((resolve, reject) => {
+        fileReadStream.on('end', () => {
+          logMemory('ADDED', `File ${fileCount + 1}: ${fullFileName}`);
+          resolve();
+        });
+        fileReadStream.on('error', (err: any) => {
+          console.error(`[ZIP] Stream error for ${fullFileName}:`, err);
+          reject(err);
+        });
+      });
+
+      promises.push(streamPromise);
+    }
+  }
+
+  return promises;
+}
+
+/**
+ * Logs current memory usage during development and debugging.
+ *
+ * Memory metrics logged:
+ * - RSS: Total memory allocated to the process
+ * - Heap Used/Total: JavaScript heap memory usage
+ * - ArrayBuffers: Memory used by ArrayBuffers (file buffers)
+ * - External: Memory used by external C++ objects
+ *
+ * Only active when DEVELOPER_MODE is enabled to avoid performance overhead
+ * in production.
+ *
+ * @param stage - Label for this logging checkpoint
+ * @param extraInfo - Additional context information
+ */
+function logMemory(stage: string, extraInfo = ''): void {
+  if (!DEVELOPER_MODE) return;
```

Comment:
In general we don't want this in developer mode all the time, note sure it's worth keeping beyond this PR but if you want to leave it in can you add a more specific setting.